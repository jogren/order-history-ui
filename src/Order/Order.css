.Order_section {
  display: flex;
  justify-content: space-between;
  /* align-items: center; */
  background-color: #FFF;
  margin: 40px 10%;
  box-shadow: 0 0 10px #000000;
  height: 100px;
}

.Order_div-image {
  display: flex;
  justify-content: center;
  align-items: center;
}

img {
  height: 70px;
  width: 90px;
}

.Order_div-content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.Order_div-price {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

h3 {
  margin-top: 12px;
  font-size: 24px;
}

.Order_button {
  height: 50px;
  width: 100px;
  background-color: lightblue;
}
