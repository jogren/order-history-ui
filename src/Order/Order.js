import React from 'react';
import './Order.css';

const Order = ({ id, name, description, price, img, deletePurchase }) => {

  return (
    <section className="Order_section">
      <img src={img} alt={name}/>
      <div className="Order_section-div">
        <h2>{name}</h2>
        <p>{description}</p>
      </div>
      <div>
        <h3>{price}</h3>
        <button onClick={() => deletePurchase(id)}>Remove from History</button>
      </div>
    </section>
  )
}

export default Order;