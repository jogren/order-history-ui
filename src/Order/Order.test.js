import React from 'react';
import { shallow } from 'enzyme';
import Order from './Order';

describe('Order', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
    <Order 
      id={1} 
      name="iPhone" 
      description="The new iPhone18" 
      img="https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154"
      price={200}
      deletePurchase={jest.fn()}
    />)
  })

  it('should keep an up-to-date snapshot of Order', () => {
    expect(wrapper).toMatchSnapshot();
  })
})