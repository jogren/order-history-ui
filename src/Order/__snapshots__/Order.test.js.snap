// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should keep an up-to-date snapshot of Order 1`] = `
<section
  className="Order_section"
>
  <div
    className="Order_div-image"
  >
    <img
      alt="iPhone"
      src="https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154"
    />
  </div>
  <div
    className="Order_div-content"
  >
    <h2>
      iPhone
    </h2>
    <p>
      The new iPhone18
    </p>
  </div>
  <div
    className="Order_div-price"
  >
    <h3>
      $
      200
    </h3>
    <button
      className="Order_button"
      onClick={[Function]}
    >
      Remove from History
    </button>
  </div>
</section>
`;
